maint.php

```diff
@@ -819,57 +845,235 @@ function applyFilter() {
 		loadPageNoHeader(strURL);
 	}
+
+	$(function() {
+		$('#rows, #site_id, #poller_id, #location, #host_template_id, #filter, #associated').change(function() {
+			applyFilter();
+		});
+
+		$('#clear').click(function() {
+			clearFilter();
+		});
+
+		$('#form_devices').submit(function(event) {
+			event.preventDefault();
+			applyFilter();
+		});
+	});
 	</script>
 	<?php
 
-	html_start_box(__('Associated Devices %s', htmlspecialchars($header_label), 'maint'), '100%', '', '3', 'center', '');
+	html_start_box(__('Associated Devices %s', $header_label), '100%', '', '3', 'center', '');
 
 	?>
 	<tr class='even'>
 		<td>
-		<form name='form_devices' method='post' action='maint.php?action=edit&tab=hosts'>
+		<form id='form_devices' action='maint.php?action=edit&tab=hosts'>
 			<table class='filterTable'>
 				<tr>
 					<td>
-						<?php print __('Search', 'maint');?>
+						<?php print __('Site');?>
+					</td>
+					<td>
+						<select id='site_id'>
+							<option value='<?php print HOST_FILTER_ANY ?>' <?php if (get_request_var('site_id') == HOST_FILTER_ANY) {?> selected<?php }?>><?php print __('Any');?></option>
+							<option value='<?php print HOST_FILTER_NONE ?>' <?php if (get_request_var('site_id') == HOST_FILTER_NONE) {?> selected<?php }?>><?php print __('None');?></option>
+							<?php
+							$sites = db_fetch_assoc('SELECT id, name
+								FROM sites
+								WHERE id IN (SELECT site_id FROM host)
+								ORDER BY name');
+							if (cacti_sizeof($sites)) {
+								foreach ($sites as $site) {
+									print "<option value='" . $site['id'] . "'"; if (get_request_var('site_id') == $site['id']) { print ' selected'; } print '>' . html_escape($site['name']) . "</option>";
+								}
+							}
+							?>
+						</select>
+					</td>
+					<td>
+						<?php print __('Data Collector');?>
+					</td>
+					<td>
+						<select id='poller_id'>
+							<option value='<?php print HOST_FILTER_ANY ?>' <?php if (get_request_var('poller_id') == HOST_FILTER_ANY) {?> selected<?php }?>><?php print __('Any');?></option>
+							<?php
+							$pollers = db_fetch_assoc("SELECT id, name
+								FROM poller
+								ORDER BY name");
+
+							if (cacti_sizeof($pollers)) {
+								foreach ($pollers as $poller) {
+									print "<option value='" . $poller['id'] . "'"; if (get_request_var('poller_id') == $poller['id']) { print ' selected'; } print '>' . html_escape($poller['name']) . "</option>";
+								}
+							}
+							?>
+						</select>
 					</td>
 					<td>
-						<input type='text' id='filter' size='25' value='<?php print htmlspecialchars(get_request_var('filter'));?>' onChange='applyFilter()'>
+						<?php print __('Location');?>
+					</td>
+					<td>
+						<select id='location'>
+							<option value='<?php print HOST_FILTER_LOC_ANY ?>' <?php if (get_request_var('location') == HOST_FILTER_LOC_ANY) {?> selected<?php }?>><?php print __('Any');?></option>
```

Comment:
Rather than explicitly output Any as a location.  Why not add it to the locations array further down so the same piece of code is used for all output.  That way, if there are any changes to the URL or display of it, they are all handled at once?

Reply:
That code was misplaced and not working.  Thank you.  It is added to the SQL; I do not know how to add it to the array.